decisions/inexact-rest.md

```diff
@@ -0,0 +1,56 @@
+# Inexact ...rest and TypeScript
+
+## TL;DR
+
+If using `...rest` parameter with the intention to allow any arbitrary prop to be passed then type using:
+
+```tsx
+type Props = {
+  namedPropOne: string;
+  namedPropTwo: boolean;
+  [rest: string]: any; // Inexact rest. See decisions/inexact-rest.md
```

Comment:
I would always default to stricter recommendation first, like mentioned below: `React.HTMLAttributes<HTMLDivElement>;`. I don't know all the cases, but IMO they are never arbitrary and spread to a specific element that has specific list of supported attributes.